### PR TITLE
[enhancement] Enable oneDAL support for EmpiricalCovariance with default parameters

### PR DESCRIPTION
**Pull request by icfaust**
# Description
EmpiricalCovariance is missing certain attributes necessary to run oneDAL algorithms for default parameters. This limits its utility. By applying sklearn EmpiricalCovariance native functions, the default output can be supported. This will enhance user experience, and improve testability of the preview algorithm.

Changes proposed in this pull request:
- add store_precision to EmpiricalCovariance so that default parameters will be supported
- changes to patching conditions change to reflect new capabilities
- wrap error_norm and score methods for dpnp and dpctl support
- Add mahalanobis using sklearnex's pairwise_distances (if metric implemented in the future)

### CODE REVIEW

File: sklearnex/preview/covariance/covariance.py [hunk 1]
```diff
@@ -39,7 +42,7 @@ class EmpiricalCovariance(sklearn_EmpiricalCovariance):
 
     def _save_attributes(self):
         assert hasattr(self, "_onedal_estimator")
-        self.covariance_ = self._onedal_estimator.covariance_
+        self._set_covariance(self._onedal_estimator.covariance_)
```
Comment (olegkkruglov) [inline sklearnex/preview/covariance/covariance.py:45]: where is definition of this method?

Comment (icfaust) [reply sklearnex/preview/covariance/covariance.py:45]: https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/covariance/_empirical_covariance.py#L195

File: sklearnex/preview/covariance/covariance.py [hunk 2]
```diff
@@ -100,4 +106,27 @@ def fit(self, X, y=None):
 
         return self
 
+    # This needs to be included to guarantee that we use the sklearnex pairwise_distances
```
Comment (olegkkruglov) [inline sklearnex/preview/covariance/covariance.py:109]: why is this necessary? we don't support this metric in sklearnex anyway, so, the default one still will be called.

File: sklearnex/preview/covariance/covariance.py [hunk 3]
```diff
@@ -100,4 +106,27 @@ def fit(self, X, y=None):
...
+    @wrap_output_data
+    def mahalanobis(self, X, queue=None):
```
Comment (samir-nasibli) [inline sklearnex/preview/covariance/covariance.py:111]: Could you please clarify, why queue is here?

File: sklearnex/preview/covariance/covariance.py [hunk 4]
```diff
@@ -100,4 +106,27 @@ def fit(self, X, y=None):
...
+        with config_context(assume_finite=True):
+            # compute mahalanobis distances
+            dist = pairwise_distances(
+                X, self.location_[np.newaxis, :], metric="mahalanobis", VI=precision
+            )
+
+        return np.reshape(dist, (len(X),)) ** 2
+
+    error_norm = wrap_output_data(sklearn_EmpiricalCovariance.error_norm)
+    score = wrap_output_data(sklearn_EmpiricalCovariance.score)
```
Comment (samir-nasibli) [inline sklearnex/preview/covariance/covariance.py:127]: It make sense generally have a mechanism, that wraps output data for all non-patched method of the estimator.

File: sklearnex/preview/covariance/covariance.py (hunk 3 above)
Comment (samir-nasibli) [reply sklearnex/preview/covariance/covariance.py:111]: I don't like this API inconsistency for `fit` and `mahalanobis`

File: sklearnex/preview/covariance/covariance.py (hunk 4 above)
Comment (icfaust) [reply sklearnex/preview/covariance/covariance.py:127]: Agreed, this should be a follow up PR which we will have to add to the backlog (I imagine it will be a long undertaking)

File: sklearnex/preview/covariance/covariance.py (hunk 3 above)
Comment (icfaust) [reply sklearnex/preview/covariance/covariance.py]: I have removed the queue. pairwise_distances in daal4py has input checks within it following daal4py style.

**Review by olegkkruglov — APPROVED** [review]
LGTM
